### PR TITLE
ci: cache pre-commit

### PR DESCRIPTION
**Pull request by jsolaas**

### CODE REVIEW

File: .github/workflows/pre-ci.yml [hunk 1]
```diff
@@ -18,9 +18,13 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: "3.8"
+          cache: "pip"
 
       - name: Install pre-commit
-        run: pip install pre-commit==3.2.2
-
+        run: python -m pip install -r pre-commit-requirements.txt
```
Comment (TeeeJay) [inline .github/workflows/pre-ci.yml:24]: why not pip directly but through module? required?

File: .github/workflows/pre-ci.yml [hunk 2]
```diff
@@ -18,9 +18,13 @@ jobs:
...
+      - uses: actions/cache@v4
+        with:
+          path: ~/.cache/pre-commit/
+          key: pre-commit-3|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
```
Comment (TeeeJay) [inline .github/workflows/pre-ci.yml:28]: is the cache shared across actions and workflows - so it needs to be unique? that is why you add a -3- to the key? risk of using incorrect cache?

**Review by TeeeJay — APPROVED** [review]

File: .github/workflows/pre-ci.yml (hunk 1 above)
Comment (jsolaas) [reply .github/workflows/pre-ci.yml:24]: Isn't it safer? You are guaranteed that you are using the correct pip/module

File: .github/workflows/pre-ci.yml (hunk 2 above)
Comment (jsolaas) [reply .github/workflows/pre-ci.yml:28]: Shared yes, added a 3 because of pre-commit version. Not sure if it's necessary.

**Pull request merged**